src/tests/resistance/handshake.rs

```diff
@@ -212,3 +215,185 @@ async fn r001_t5_HANDSHAKE_ws_version() {
     let cfg = gen_cfg_huge(0);
     assert!(!run_handshake_req_test_with_cfg(cfg, false).await);
 }
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn r001_t6_HANDSHAKE_tel_id() {
+    // ZG-RESISTANCE-001
+
+    let gen_cfg = |len| HandshakeCfg {
+        ar_tel_id: Some(gen_huge_string(len)),
+        ..Default::default()
+    };
+
+    // Valid scenarios:
+
+    // Find the largest instance value which the node can accept.
+    let cfg = gen_cfg(WS_HTTP_HEADER_MAX_SIZE);
+    assert!(run_handshake_req_test_with_cfg(cfg, false).await);
+
+    // Send an empty field.
+    let cfg = gen_cfg(0);
+    assert!(run_handshake_req_test_with_cfg(cfg, false).await);
+
+    // Below tests assert the connection shouldn't be established.
+
+    // Use a huge value which the node will reject.
+    let cfg = gen_cfg(WS_HTTP_HEADER_INVALID_SIZE);
+    assert!(!run_handshake_req_test_with_cfg(cfg, false).await);
+}
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn r001_t7_HANDSHAKE_ws_key() {
+    // ZG-RESISTANCE-001
+
+    let gen_cfg = |len| -> HandshakeCfg {
+        let mut ws_key = SecWebSocket::generate();
+        ws_key.key = gen_huge_string(len);
+
+        HandshakeCfg {
+            ws_key: Some(ws_key),
+            ..Default::default()
+        }
+    };
+
+    // Below tests assert the connection shouldn't be established.
+
+    // Find the largest instance value which the node can accept.
+    let cfg = gen_cfg(WS_HTTP_HEADER_MAX_SIZE);
+    assert!(!run_handshake_req_test_with_cfg(cfg, false).await);
+
+    // Send an empty field.
+    let cfg = gen_cfg(0);
+    assert!(!run_handshake_req_test_with_cfg(cfg, false).await);
+
+    // Use a huge value which the node will reject.
+    let cfg = gen_cfg(WS_HTTP_HEADER_INVALID_SIZE);
+    assert!(!run_handshake_req_test_with_cfg(cfg, false).await);
+}
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn r001_t8_HANDSHAKE_location() {
+    // ZG-RESISTANCE-001
+
+    let gen_cfg = |len| HandshakeCfg {
+        ar_location: Some(gen_huge_string(len)),
+        ..Default::default()
+    };
+
+    // Below tests assert the connection shouldn't be established.
+
+    // NOTE: The node should reject an invalid address.
```

Comment:
Below tests are commented - maybe leave a TODO to not forget to get back here in the future.

Reply:
We'll check these Notes once we go through this with stakeholders. I feel a `TODO` should be used when there is something to do, and we should separate those from these questionable observations we have here